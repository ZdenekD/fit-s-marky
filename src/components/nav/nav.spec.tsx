import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../__test__/utils/helpers';
import Nav from '.';

jest.mock('next/router', () => ({useRouter: jest.fn().mockImplementation(() => ({pathname: '/'}))}));
jest.mock('framer-motion', () => {
    const motion = {
        nav: jest.fn().mockImplementation(({children}) => (
            <nav className="nav" data-test="component-nav">{children}</nav>
        )),
        ul: jest.fn().mockImplementation(({children}) => (
            <ul className="list">{children}</ul>
        )),
        li: jest.fn().mockImplementation(({children}) => (
            <li className="item ">{children}</li>
        )),
    };

    return {motion};
});

describe('Nav', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        React.useState = jest.fn(() => [true, jest.fn()]);

        wrapper = shallow(<Nav />);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-nav');

        expect(component.exists()).toBe(true);
    });
});
