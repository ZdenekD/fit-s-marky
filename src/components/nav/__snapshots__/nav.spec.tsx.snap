// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav match snapshot 1`] = `
<nav class="nav"
     data-test="component-nav"
>
  <ul class="list">
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-karate"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <circle cx="18"
                cy="4"
                r="1"
        >
        </circle>
        <path d="M3 9l4.5 1l3 2.5">
        </path>
        <path d="M13 21v-8l3 -5.5">
        </path>
        <path d="M8 4.5l4 2l4 1l4 3.5l-2 3.5">
        </path>
      </svg>
      <a href="/lekce-pilates"
         class="anchor link "
         data-test="component-anchor"
      >
        Lekce pilates
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-woman"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <circle cx="12"
                cy="5"
                r="2"
        >
        </circle>
        <path d="M10 22v-4h-2l2 -6a1 1 0 0 1 1 -1h2a1 1 0 0 1 1 1l2 6h-2v4">
        </path>
      </svg>
      <a href="/lekce-jogy-pro-tehotne"
         class="anchor link "
         data-test="component-anchor"
      >
        Lekce jógy pro těhotné
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-certificate"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <circle cx="15"
                cy="15"
                r="3"
        >
        </circle>
        <path d="M13 17.5v4.5l2 -1.5l2 1.5v-4.5">
        </path>
        <path d="M10 19h-5a2 2 0 0 1 -2 -2v-10c0 -1.1 .9 -2 2 -2h14a2 2 0 0 1 2 2v10a2 2 0 0 1 -1 1.73">
        </path>
        <line x1="6"
              y1="9"
              x2="18"
              y2="9"
        >
        </line>
        <line x1="6"
              y1="12"
              x2="9"
              y2="12"
        >
        </line>
        <line x1="6"
              y1="15"
              x2="8"
              y2="15"
        >
        </line>
      </svg>
      <a href="/certifikaty"
         class="anchor link "
         data-test="component-anchor"
      >
        Certifikáty
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-calendar-event"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <rect x="4"
              y="5"
              width="16"
              height="16"
              rx="2"
        >
        </rect>
        <line x1="16"
              y1="3"
              x2="16"
              y2="7"
        >
        </line>
        <line x1="8"
              y1="3"
              x2="8"
              y2="7"
        >
        </line>
        <line x1="4"
              y1="11"
              x2="20"
              y2="11"
        >
        </line>
        <rect x="8"
              y="15"
              width="2"
              height="2"
        >
        </rect>
      </svg>
      <a href="/cviceni"
         class="anchor link "
         data-test="component-anchor"
      >
        Kdy a kde cvičím
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-photo"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <line x1="15"
              y1="8"
              x2="15.01"
              y2="8"
        >
        </line>
        <rect x="4"
              y="4"
              width="16"
              height="16"
              rx="3"
        >
        </rect>
        <path d="M4 15l4 -4a3 5 0 0 1 3 0l5 5">
        </path>
        <path d="M14 14l1 -1a3 5 0 0 1 3 0l2 2">
        </path>
      </svg>
      <a href="/fotogalerie"
         class="anchor link "
         data-test="component-anchor"
      >
        Fotogalerie
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-receipt"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <path d="M5 21v-16a2 2 0 0 1 2 -2h10a2 2 0 0 1 2 2v16l-3 -2l-2 2l-2 -2l-2 2l-2 -2l-3 2m4 -14h6m-6 4h6m-2 4h2">
        </path>
      </svg>
      <a href="/cenik"
         class="anchor link "
         data-test="component-anchor"
      >
        Ceník
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-physotherapist"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <path d="M9 15l-1 -3l4 -2l4 1h3.5">
        </path>
        <circle cx="4"
                cy="19"
                r="1"
        >
        </circle>
        <circle cx="12"
                cy="6"
                r="1"
        >
        </circle>
        <path d="M12 17v-7">
        </path>
        <path d="M8 20h7l1 -4l4 -2">
        </path>
        <path d="M18 20h3">
        </path>
      </svg>
      <a href="/klienti"
         class="anchor link "
         data-test="component-anchor"
      >
        Klienti
      </a>
    </li>
    <li class="item ">
      <svg class="icon icon-tabler icon-tabler-message"
           width="28"
           height="28"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <path d="M4 21v-13a3 3 0 0 1 3 -3h10a3 3 0 0 1 3 3v6a3 3 0 0 1 -3 3h-9l-4 4">
        </path>
        <line x1="8"
              y1="9"
              x2="16"
              y2="9"
        >
        </line>
        <line x1="8"
              y1="13"
              x2="14"
              y2="13"
        >
        </line>
      </svg>
      <a href="/kontakt"
         class="anchor link "
         data-test="component-anchor"
      >
        Kontakt
      </a>
    </li>
  </ul>
</nav>
`;
